tests/cxx/typedef_chain_in_template-d4.h

```diff
@@ -0,0 +1,25 @@
+//===--- typedef_chain_in_template-d4.h - test input file for iwyu --------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+// Container that tests typedef chains longer than in libc++.
+
+template<typename T>
+class ContainerLongTypedefChain {
+ public:
+  typedef T value_type;
+  typedef value_type& reference;
+  typedef reference& reference_reference;
```

Comment:
I think this is reference collapsing -- http://stackoverflow.com/a/3772234/96963 -- if it affects IWYU analysis it's definitely a good thing to cover.